File: kotlin/605_Can_Place_Flowers.kt
class Solution {
    fun canPlaceFlowers(flowerbed: IntArray, n: Int): Boolean {
        var countFlowers = n;
        for (i in 0 until flowerbed.size) {
            if (
                flowerbed[i] == 0 
                && (i == 0 || flowerbed[i - 1] == 0)
                && (i == flowerbed.size - 1 || flowerbed[i + 1] == 0)
            ) {
                flowerbed[i] = 1;
                countFlowers--;
            }
        }

        return countFlowers <= 0;
    }
}

File: scala/605_Can_Place_Flowers.scala
object Solution {
    def canPlaceFlowers(flowerbed: Array[Int], n: Int): Boolean = {
        var countFlowers = n;
        for (i <- 0 to (flowerbed.length - 1)) {
            if (
                flowerbed(i) == 0 
                && (i == 0 || flowerbed(i - 1) == 0)
                && (i == flowerbed.length - 1 || flowerbed(i + 1) == 0)
            ) {
                flowerbed(i) = 1;
                countFlowers -= 1;
            }
        }
        return countFlowers <= 0;
    }
}
